### PR TITLE
udpate some packages and resolve conflicts

### PR DESCRIPTION
**Pull request by dtebbs**
Fixes some issues with package compatibility.
Tested on mac and on a docker container based on `ghcr.io/clearmatics/zeth:0.0.2-base`.

### CODE REVIEW

**Comment by AntoineRondelet** [comment]
Addresses https://github.com/clearmatics/zeth/issues/398

File: Cargo.toml [hunk 1]
```diff
@@ -13,10 +13,10 @@ repository = "https://github.com/ebfull/powersoftau"
 rand = { version = "0.5", features = ["i128_support"] }
 crossbeam = "0.3.0"
 num_cpus = "1.7.0"
-blake2 = "0.6.1"
-generic-array = "0.8.3"
+blake2 = "0.9.1"
+generic-array = "0"
 typenum = "1.9.0"
-byteorder = { version = "1.0", features = ["i128"] }
+byteorder = "1"
```
Comment (AntoineRondelet) [inline Cargo.toml]: Why do we build it without the i128 feature now? https://docs.rs/byteorder/1.4.3/byteorder/#optional-features

Comment (dtebbs) [reply Cargo.toml]: I've added `i128` back in.

Comment (AntoineRondelet) [reply Cargo.toml]: Thanks. Was wondering why we needed it in the first place. As discussed, it doesn't seem necessary. In fact, by looking a the code of the crate  https://github.com/BurntSushi/byteorder/blob/1.4.3/src/lib.rs they don't seem to use the i128  feature (I can't see any occurence of `#[cfg(feature = "i128")]`). Looking into the changelog of the project (see https://github.com/BurntSushi/byteorder/blob/master/CHANGELOG.md#130) we see the following:
> This new minor release now enables i128 support automatically on Rust compilers that support 128-bit integers. The i128 feature is now a no-op, but continues to exist for backward compatibility purposes. The crate continues to maintain compatibility with Rust 1.12.0.

So this feature seems to be kept simply for backward compatibility with old rust toolkits AFAICT (see: https://github.com/BurntSushi/byteorder/issues/65).

**Review by AntoineRondelet — APPROVED** [review]
LGTM, thanks for fixing this @dtebbs

**Pull request merged**